apps/mobile/src/features/send/send-form/providers/send-form-stx-provider.tsx

```diff
@@ -0,0 +1,63 @@
+import { useGetStacksAddresses } from '@/features/balances/stacks/use-get-stacks-addresses';
+import { useStxBalance } from '@/queries/balance/stacks-balance.query';
+import { HasChildren } from '@/utils/types';
+import { t } from '@lingui/macro';
+
+import { defaultStacksFees, useNextNonce } from '@leather.io/query';
+import { convertAmountToBaseUnit, createMoney } from '@leather.io/utils';
+
+import { CreateCurrentSendRoute, useSendSheetRoute } from '../../send-form.utils';
+import { useSendFormStx } from '../hooks/use-send-form-stx';
+import { defaultSendFormStxValues, sendFormStxSchema } from '../schemas/send-form-stx.schema';
+import { SendFormProvider } from '../send-form-context';
+
+const defaultFeeFallback = 2500;
+
+type CurrentRoute = CreateCurrentSendRoute<'send-form-stx'>;
+
+export function SendFormStxProvider({ children }: HasChildren) {
+  const route = useSendSheetRoute<CurrentRoute>();
+
+  const { onSubmit } = useSendFormStx();
+
+  const addresses = useGetStacksAddresses({
+    accountIndex: route.params.account.accountIndex,
+    fingerprint: route.params.account.fingerprint,
+  });
+  const { availableBalance, fiatBalance } = useStxBalance(addresses);
+
+  const address = addresses[0] ?? '';
+  const { data: nextNonce } = useNextNonce(address);
+
+  // Load fees here, temp default fee
+  const defaultFeeFallbackAsMoney = createMoney(defaultFeeFallback, 'STX');
+  const defaultFee = convertAmountToBaseUnit(
+    defaultStacksFees.estimates[0]?.fee ?? defaultFeeFallbackAsMoney
+  );
+
+  if (!nextNonce) return null;
```

Comment:
Should we have a `nonce` loader component, or even a loader component specific to this send form? 

I could see the Loader pattern living side by side use of context here.

```ts
<StxSendFormLoader fallback={<LoadingScreen />}>
  {nonceEtc => <SendFormProvider {...nonceEtc} />}
</StxSendFormLoader/>
```

Reply:
Yep, agree, was refactoring to this when you were reviewing. I was thinking maybe we only need loaders per chain, but maybe we do need per protocol. 🤔 Stacks would load the same fees and nonce for stx and sip10, so there would be some duplication there if do separately.

Reply:
Thinking should be separate, so separating. I think the loader pattern makes more sense to me rather than using the hook for loading, but open if others feel strongly either way.